vault CLI client 0.11.4 errors using aws login

**To Reproduce**

```
% env | grep VAULT
VAULT_CACERT=/Users/me/.certs/vault-ca.pem
VAULT_ADDR=https://myvaultserver.com:8200/
% vault --version
Vault v0.11.4 ('612120e76de651ef669c9af5e77b27a749b0dba3')
% vault login -method aws
panic: assignment to entry in nil map
goroutine 1 [running]:
github.com/hashicorp/vault/command.(*LoginCommand).Run(0xc0001fc280, 0xc0000960a0, 0x0, 0x0, 0xc0004d0060)
	/private/tmp/vault-20181024-94467-1i61kxt/src/github.com/hashicorp/vault/command/login.go:255 +0xdee
github.com/hashicorp/vault/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc000523540, 0xc000523540, 0xc0004d04a0, 0xc00024e5a0)
	/private/tmp/vault-20181024-94467-1i61kxt/src/github.com/hashicorp/vault/vendor/github.com/mitchellh/cli/cli.go:255 +0x207
github.com/hashicorp/vault/command.RunCustom(0xc000096090, 0x3, 0x3, 0xc0000b2b90, 0x0)
	/private/tmp/vault-20181024-94467-1i61kxt/src/github.com/hashicorp/vault/command/main.go:170 +0x74a
github.com/hashicorp/vault/command.Run(0xc000096090, 0x3, 0x3, 0xc000088058)
	/private/tmp/vault-20181024-94467-1i61kxt/src/github.com/hashicorp/vault/command/main.go:81 +0x48
main.main()
	/private/tmp/vault-20181024-94467-1i61kxt/src/github.com/hashicorp/vault/main.go:10 +0x62
```

**Expected behavior**
I expect the behavior from 0.11.1

```
% vault login -method aws
Success! You are now authenticated. The token information displayed below
is already stored in the token helper. You do NOT need to run "vault login"
again. Future Vault requests will automatically use this token.

Key                                Value
---                                -----
...
```

**Environment:**
* Vault Server Version (retrieve with `vault status`):
```
% vault status
Key             Value
---             -----
Seal Type       shamir
Sealed          false
Total Shares    8
Threshold       2
Version         0.10.0
Cluster Name    vault-cluster-name
Cluster ID      uuid
HA Enabled      true
HA Cluster      https://x.x.x.x:8201
HA Mode         active
```
* Vault CLI Version (retrieve with `vault version`):
0.11.4

* Server Operating System/Architecture:
osx installing vault cli via home brew `brew install vault`

I wanted to make sure this wasn't just an OSX or homebrew issue, so I reproduced it on one of our linux server

```
bash-4.3# uname -a
Linux myapp-77b5fdf5b9-rsxsp 4.14.67-coreos #1 SMP Mon Sep 10 23:14:26 UTC 2018 x86_64 Linux
bash-4.3# wget https://releases.hashicorp.com/vault/0.11.4/vault_0.11.4_linux_amd64.zip
bash-4.3# unzip vault_0.11.4_linux_amd64.zip -d /usr/local/bin
bash-4.3# vault --version
Vault v0.11.4 ('612120e76de651ef669c9af5e77b27a749b0dba3')
bash-4.3# vault login -method aws
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/hashicorp/vault/command.(*LoginCommand).Run(0xc0000c07d0, 0xc0000b6020, 0x0, 0x0, 0xc000227a00)
        /gopath/src/github.com/hashicorp/vault/command/login.go:255 +0xdee
github.com/hashicorp/vault/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc000546dc0, 0xc000546dc0, 0xc000227dc0, 0xc00023c780)
        /gopath/src/github.com/hashicorp/vault/vendor/github.com/mitchellh/cli/cli.go:255 +0x207
github.com/hashicorp/vault/command.RunCustom(0xc0000b6010, 0x3, 0x3, 0xc0006a8640, 0x0)
        /gopath/src/github.com/hashicorp/vault/command/main.go:170 +0x74a
github.com/hashicorp/vault/command.Run(0xc0000b6010, 0x3, 0x3, 0xc0000a0058)
        /gopath/src/github.com/hashicorp/vault/command/main.go:81 +0x48
main.main()
        /gopath/src/github.com/hashicorp/vault/main.go:10 +0x62
```

--- a/command/base_helpers.go
+++ b/command/base_helpers.go
@@ -149,6 +149,9 @@ func parseArgsDataString(stdin io.Reader, args []string) (map[string]string, err
 	if err := mapstructure.WeakDecode(raw, &result); err != nil {
 		return nil, errors.Wrap(err, "failed to convert values to strings")
 	}
+	if result == nil {
+		result = make(map[string]string)
+	}
 	return result, nil
 }
 

